app/src/main/java/com/kickstarter/services/KSApolloClientV2.kt

```diff
@@ -896,4 +897,45 @@ class KSApolloClientV2(val service: ApolloClient) : ApolloClientTypeV2 {
             return@defer ps
         }.subscribeOn(Schedulers.io())
     }
+
+    override fun fetchCategories(): Observable<List<Category>> {
+        return Observable.defer {
+            val ps = PublishSubject.create<List<Category>>()
+            this.service.query(
+                GetRootCategoriesQuery.builder()
+                    .build()
+            ).enqueue(object : ApolloCall.Callback<GetRootCategoriesQuery.Data>() {
+                override fun onFailure(e: ApolloException) {
+                    ps.onError(e)
+                }
+
+                override fun onResponse(response: Response<GetRootCategoriesQuery.Data>) {
+                    if (response.hasErrors()) {
+                        ps.onError(Exception(response.errors?.first()?.message))
+                    } else {
+                        response.data?.let { responseData ->
+                            val subCategories = responseData.rootCategories()
+                                .flatMap { it.subcategories()?.nodes().orEmpty() }
+                                .map {
+                                    categoryTransformer(it.fragments().category())
+                                }
+                            val rootCategories = responseData.rootCategories()
+                                .map { categoryTransformer(it.fragments().category()) }
+                                .toMutableList()
+                                .apply {
+                                    addAll(subCategories)
+                                }
+                            Observable.just(rootCategories)
```

Comment:
Suggestion, not mandatory, but we could avoid this observable subscription and do directly `ps.onNext(rootCategories)`